core/templates/pages/diagnostic-test-player-page/diagnostic-test-topic-state.model.ts

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Diagnostic test topic state model.
+ */
+
+import { Question } from 'domain/question/QuestionObjectFactory';
+
+export interface SkillIdToQuestions {
+  [skillId: string]: Question[];
+}
+
+export class DiagnosticTestTopicStateData {
```

Comment:
Why is the word "topic" needed in the name (what does it convey and what misinterpretation is possible if that word is removed from the name)?

Reply:
Renamed as per discussion